app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt

```diff
@@ -95,39 +115,42 @@ class EventViewModel() : ViewModel() {
         } else {
             if (eventIsValid()) {
                 _status.value = EventStatus.Saving
-                // TODO: save in repository
+                viewModelScope.launch {
+                    if (isEventNew.value) {
+                        // TODO: get new eventId for new events (not implemented yet in the
+                        // repository)
+                    } else {
+                        repository.setEvent(_event.value)
+                    }
+                    _isEventNew.value = false
+                    _status.value = EventStatus.Saved
+                }
             }
         }
     }
 
-    // return the status of the event
-    fun getStatus(): EventStatus {
-        return _status.asStateFlow().value
-    }
-
     /** check the current event has valid data if not return false and set _status to Error */
     private fun eventIsValid(): Boolean {
         val event = _event.value
         if (event.startDate.isAfter(event.endDate)) {
-            _status.value = EventStatus.Error("end date before start date")
+            _status.value =
+                EventStatus.Error(R.string.event_creation_error_end_date_before_start_date)
         }
         if (event.title.isBlank()) {
-            _status.value = EventStatus.Error("title is empty")
+            _status.value = EventStatus.Error(R.string.event_creation_error_empty_title)
         }
         return _status.value !is EventStatus.Error
     }
 }
 
 /** the different status of an event */
 sealed class EventStatus {
-    // new event not in the repository
-    data object New : EventStatus()
     // same as in the repository
     data object Saved : EventStatus()
     // different from the version in the repository
     data object Modified : EventStatus()
     // syncing with the version in the repository
     data object Saving : EventStatus()
```

Comment:
Nitpicking: maybe `Syncing` would be more clear

Reply:
I prefer to keep `Saving` for now, because the button in the UI is called "Save". Maybe when add the edit event screen we could rename it.